<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>An overview of tRavis • tRavis</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An overview of tRavis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tRavis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/tRavis.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/travis-m-blimkie/tRavis/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>An overview of tRavis</h1>
                        <h4 data-toc-skip class="author">Travis
Blimkie</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/travis-m-blimkie/tRavis/blob/master/vignettes/tRavis.Rmd" class="external-link"><code>vignettes/tRavis.Rmd</code></a></small>
      <div class="d-none name"><code>tRavis.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://travis-m-blimkie.github.io/tRavis/">tRavis</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Thanks for using tRavis v1.6.0! If you encounter any bugs</span></span>
<span><span class="co">#&gt; or problems, please submit an issue at the Github page:</span></span>
<span><span class="co">#&gt; https://github.com/travis-m-blimkie/tRavis/issues</span></span></code></pre></div>
<div class="section level2">
<h2 id="tr_anno_cleaner">tr_anno_cleaner<a class="anchor" aria-label="anchor" href="#tr_anno_cleaner"></a>
</h2>
<p>Clean annotation files (CSV or TSV) for <em>Pseudomonas
aeruginosa</em> from <a href="https://pseudomonas.com" class="external-link uri">https://pseudomonas.com</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># A URL can also be supplied, instead of a local file path</span></span>
<span><span class="va">link</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>  <span class="st">"https://pseudomonas.com/downloads/pseudomonas/pgd_r_22_1/"</span>,</span>
<span>  <span class="st">"Pseudomonas_aeruginosa_PAO1_107/Pseudomonas_aeruginosa_PAO1_107.csv.gz"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/tr_anno_cleaner.html">tr_anno_cleaner</a></span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">link</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,713 × 3</span></span></span>
<span><span class="co">#&gt;    locus_tag gene_name product_name                                  </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PA0001    dnaA      chromosomal replication initiator protein DnaA</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PA0002    dnaN      DNA polymerase III, beta chain                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PA0003    recF      RecF protein                                  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PA0004    gyrB      DNA gyrase subunit B                          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PA0005    lptA      lysophosphatidic acid acyltransferase, LptA   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PA0006    <span style="color: #BB0000;">NA</span>        conserved hypothetical protein                </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PA0007    <span style="color: #BB0000;">NA</span>        hypothetical protein                          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PA0008    glyS      glycyl-tRNA synthetase beta chain             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PA0009    glyQ      glycyl-tRNA synthetase alpha chain            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PA0010    tag       DNA-3-methyladenine glycosidase I             </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5,703 more rows</span></span></span></code></pre></div>
<p>Or add some extra columns and fill empty names with the corresponding
locus tag.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tr_anno_cleaner.html">tr_anno_cleaner</a></span><span class="op">(</span><span class="va">link</span>, extra_cols <span class="op">=</span> <span class="cn">TRUE</span>, fill_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5,713 × 6</span></span></span>
<span><span class="co">#&gt;    locus_tag gene_name product_name                           start   end strand</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> PA0001    dnaA      chromosomal replication initiator pro…   483  <span style="text-decoration: underline;">2</span>027 +     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> PA0002    dnaN      DNA polymerase III, beta chain          <span style="text-decoration: underline;">2</span>056  <span style="text-decoration: underline;">3</span>159 +     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> PA0003    recF      RecF protein                            <span style="text-decoration: underline;">3</span>169  <span style="text-decoration: underline;">4</span>278 +     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> PA0004    gyrB      DNA gyrase subunit B                    <span style="text-decoration: underline;">4</span>275  <span style="text-decoration: underline;">6</span>695 +     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> PA0005    lptA      lysophosphatidic acid acyltransferase…  <span style="text-decoration: underline;">7</span>018  <span style="text-decoration: underline;">7</span>791 -     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> PA0006    PA0006    conserved hypothetical protein          <span style="text-decoration: underline;">7</span>803  <span style="text-decoration: underline;">8</span>339 -     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> PA0007    PA0007    hypothetical protein                    <span style="text-decoration: underline;">8</span>671 <span style="text-decoration: underline;">10</span>377 +     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> PA0008    glyS      glycyl-tRNA synthetase beta chain      <span style="text-decoration: underline;">10</span>434 <span style="text-decoration: underline;">12</span>488 -     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PA0009    glyQ      glycyl-tRNA synthetase alpha chain     <span style="text-decoration: underline;">12</span>488 <span style="text-decoration: underline;">13</span>435 -     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> PA0010    tag       DNA-3-methyladenine glycosidase I      <span style="text-decoration: underline;">13</span>540 <span style="text-decoration: underline;">14</span>091 +     </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5,703 more rows</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_clean_deseq2_result">tr_clean_deseq2_result<a class="anchor" aria-label="anchor" href="#tr_clean_deseq2_result"></a>
</h2>
<p>Takes a <a href="https://bioconductor.org/packages/DESeq2/" class="external-link">DESeq2</a> results
object, and returns the significant DE genes, printing a message
summarizing the comparison and number of significant genes when
<code>inform = TRUE</code> (default).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex_deseq_results</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ex_deseq_results.rds"</span>, package <span class="op">=</span> <span class="st">"tRavis"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ex_deseq_results</span></span>
<span><span class="co">#&gt; log2 fold change (MLE): condition B vs A </span></span>
<span><span class="co">#&gt; Wald test p-value: condition B vs A </span></span>
<span><span class="co">#&gt; DataFrame with 200 rows and 6 columns</span></span>
<span><span class="co">#&gt;          baseMean log2FoldChange     lfcSE      stat      pvalue        padj</span></span>
<span><span class="co">#&gt;         &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; gene1      7.1530      -0.636105  0.750697 -0.847352 3.96799e-01 4.95998e-01</span></span>
<span><span class="co">#&gt; gene2    108.3659       2.872486  0.357737  8.029601 9.77899e-16 1.62983e-14</span></span>
<span><span class="co">#&gt; gene3     25.1923       1.724502  0.643106  2.681522 7.32882e-03 1.62863e-02</span></span>
<span><span class="co">#&gt; gene4    110.2302      -1.375727  0.321775 -4.275440 1.90761e-05 7.06521e-05</span></span>
<span><span class="co">#&gt; gene5     15.1829      -5.412610  0.819823 -6.602166 4.05195e-11 3.68359e-10</span></span>
<span><span class="co">#&gt; ...           ...            ...       ...       ...         ...         ...</span></span>
<span><span class="co">#&gt; gene196   1.28627      -0.654713  1.676396 -0.390548 6.96131e-01 7.86589e-01</span></span>
<span><span class="co">#&gt; gene197 591.21368       3.285941  0.278768 11.787369 4.53489e-32 4.53489e-30</span></span>
<span><span class="co">#&gt; gene198   4.67660       2.434828  0.961683  2.531839 1.13466e-02 2.33951e-02</span></span>
<span><span class="co">#&gt; gene199  14.96302      -2.313151  0.491582 -4.705521 2.53218e-06 1.17776e-05</span></span>
<span><span class="co">#&gt; gene200  17.69265       1.970386  0.602859  3.268401 1.08157e-03 3.00436e-03</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tr_clean_deseq2_result.html">tr_clean_deseq2_result</a></span><span class="op">(</span><span class="va">ex_deseq_results</span><span class="op">)</span></span>
<span><span class="co">#&gt; Found 100 DE genes for condition B vs A.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 7</span></span></span>
<span><span class="co">#&gt;    gene    baseMean log2FoldChange lfcSE   stat   pvalue     padj</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> gene95     853.            3.35 0.279  12.0  3.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-33</span> 6.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> gene197    591.            3.29 0.279  11.8  4.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-32</span> 4.53<span style="color: #949494;">e</span><span style="color: #BB0000;">-30</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> gene180    420.            3.30 0.298  11.1  1.57<span style="color: #949494;">e</span><span style="color: #BB0000;">-28</span> 1.05<span style="color: #949494;">e</span><span style="color: #BB0000;">-26</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> gene70     235.           -<span style="color: #BB0000;">2.91</span> 0.276 -<span style="color: #BB0000;">10.5</span>  5.23<span style="color: #949494;">e</span><span style="color: #BB0000;">-26</span> 2.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-24</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> gene106    886.            3.08 0.303  10.1  3.80<span style="color: #949494;">e</span><span style="color: #BB0000;">-24</span> 1.52<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> gene187     57.3          -<span style="color: #BB0000;">4.28</span> 0.428 -<span style="color: #BB0000;">10.0</span>  1.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-23</span> 4.74<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> gene26      93.9           4.62 0.473   9.77 1.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-22</span> 4.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> gene68      72.7          -<span style="color: #BB0000;">3.23</span> 0.345  -<span style="color: #BB0000;">9.37</span> 7.58<span style="color: #949494;">e</span><span style="color: #BB0000;">-21</span> 1.90<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> gene122    462.            2.63 0.307   8.58 9.79<span style="color: #949494;">e</span><span style="color: #BB0000;">-18</span> 2.18<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> gene39     191.            2.67 0.325   8.20 2.33<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span> 4.66<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span></code></pre></div>
<p>The default filters applied to the data are:
<code>padj &lt; 0.05</code> and
<code>abs(log2FoldChange) &gt; log2(1.5)</code>.</p>
</div>
<div class="section level2">
<h2 id="tr_compare_lists">tr_compare_lists<a class="anchor" aria-label="anchor" href="#tr_compare_lists"></a>
</h2>
<p>Compare two lists to find the common/unique elements, with an
optional <code>names</code> argument to apply to the results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tr_compare_lists.html">tr_compare_lists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">4</span>, <span class="fl">5</span>, <span class="fl">6</span><span class="op">)</span>, names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; $unique_A</span></span>
<span><span class="co">#&gt; [1] 1 2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $common</span></span>
<span><span class="co">#&gt; [1] 3 4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $unique_B</span></span>
<span><span class="co">#&gt; [1] 5 6</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_get_files">tr_get_files<a class="anchor" aria-label="anchor" href="#tr_get_files"></a>
</h2>
<p>Create a named list of files, easily piped into
<code>purrr::map(~read.csv(.x))</code> to generate a named list of data
frames. Supports recursive searching, custom string/pattern removal, and
date removal assuming a format like YYYYMMDD (can’t contain
punctuation/symbols).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tr_get_files.html">tr_get_files</a></span><span class="op">(</span></span>
<span>  directory <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"tRavis"</span><span class="op">)</span>,</span>
<span>  pattern <span class="op">=</span> <span class="st">"test"</span>,</span>
<span>  date <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_string <span class="op">=</span> <span class="st">"test_"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; $file1</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/tRavis/extdata/test_file1_20191231.csv"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file2</span></span>
<span><span class="co">#&gt; [1] "/home/runner/work/_temp/Library/tRavis/extdata/test_file2_20200101.csv"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_qc_plots">tr_qc_plots<a class="anchor" aria-label="anchor" href="#tr_qc_plots"></a>
</h2>
<p>Generate RNA-Seq QC plots from <a href="https://multiqc.info/" class="external-link">MultiQC</a> outputs. Currently only
supports summary plots for FastQC (Phred scores and read counts), STAR,
and HTSeq. Plots are created with <a href="https://ggplot2.tidyverse.org/" class="external-link">ggplot2</a> for simplicity. A few
arguments are provided to modify the overall font size, set the limits,
and toggle a threshold line at a given number of reads/counts:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiqc_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/tr_qc_plots_data"</span>, package <span class="op">=</span> <span class="st">"tRavis"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">multiqc_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "fastqc_per_base_sequence_quality_plot.tsv"</span></span>
<span><span class="co">#&gt; [2] "multiqc_fastqc.txt"                       </span></span>
<span><span class="co">#&gt; [3] "multiqc_htseq.txt"                        </span></span>
<span><span class="co">#&gt; [4] "multiqc_star.txt"</span></span>
<span></span>
<span><span class="va">qc_plot_output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tr_qc_plots.html">tr_qc_plots</a></span><span class="op">(</span></span>
<span>  directory <span class="op">=</span> <span class="va">multiqc_data</span>,</span>
<span>  threshold_line <span class="op">=</span> <span class="fl">5e6</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">14</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">qc_plot_output</span><span class="op">[[</span><span class="st">"plots"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; $phred_scores</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $fastqc_reads</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $star</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $htseq</span></span></code></pre></div>
<p><img src="tRavis_files/figure-html/tr_qc_plots-1.png" width="768"><img src="tRavis_files/figure-html/tr_qc_plots-2.png" width="768"><img src="tRavis_files/figure-html/tr_qc_plots-3.png" width="768"><img src="tRavis_files/figure-html/tr_qc_plots-4.png" width="768"></p>
<div class="section level3">
<h3 id="alternate-boxplots">Alternate boxplots<a class="anchor" aria-label="anchor" href="#alternate-boxplots"></a>
</h3>
<p>The bar plots work well enough for relatively few samples, but
quickly become unwieldy with lots of samples. Box plots can also be
generated using the same function as follows:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_plot_output_box</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tr_qc_plots.html">tr_qc_plots</a></span><span class="op">(</span></span>
<span>  directory <span class="op">=</span> <span class="va">multiqc_data</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"box"</span>,</span>
<span>  threshold_line <span class="op">=</span> <span class="fl">5e6</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">16</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">qc_plot_output_box</span><span class="op">[[</span><span class="st">"plots"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fastqc_reads"</span>, <span class="st">"star"</span>, <span class="st">"htseq"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; $fastqc_reads</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $star</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $htseq</span></span></code></pre></div>
<p><img src="tRavis_files/figure-html/tr_qc_plots_box-1.png" width="700"><img src="tRavis_files/figure-html/tr_qc_plots_box-2.png" width="700"><img src="tRavis_files/figure-html/tr_qc_plots_box-3.png" width="700"></p>
<p>The points can be toggled on or off using the <code>add_points</code>
argument.</p>
</div>
<div class="section level3 tabset">
<h3 class="tabset" id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h3>
<p>All the underlying tidy data is also returned, so one can easily
generate their own plots or further examine the data as desired:</p>




<ul class="nav nav-tabs" id="data" role="tablist">
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#phred-scores" id="phred-scores-tab" type="button" role="tab" aria-controls="phred-scores" aria-selected="true" class="active nav-link">Phred scores</button></li>
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#fastqc-reads" id="fastqc-reads-tab" type="button" role="tab" aria-controls="fastqc-reads" aria-selected="false" class="nav-link">FastQC reads</button></li>
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#star" id="star-tab" type="button" role="tab" aria-controls="star" aria-selected="false" class="nav-link">STAR</button></li>
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#htseq" id="htseq-tab" type="button" role="tab" aria-controls="htseq" aria-selected="false" class="nav-link">HTSeq</button></li>
</ul>
<div class="tab-content">
<div class="active  tab-pane" id="phred-scores" role="tabpanel" aria-labelledby="phred-scores-tab">

<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_plot_output</span><span class="op">[[</span><span class="st">"data"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"phred_scores"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 760 × 3</span></span></span>
<span><span class="co">#&gt;    sample    position phred_score</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> sample_01        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> sample_02        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> sample_03        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> sample_04        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> sample_05        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> sample_06        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> sample_07        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> sample_08        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> sample_09        1        35.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> sample_10        1        35.6</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 750 more rows</span></span></span></code></pre></div>
</div>
<div class="tab-pane" id="fastqc-reads" role="tabpanel" aria-labelledby="fastqc-reads-tab">

<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_plot_output</span><span class="op">[[</span><span class="st">"data"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"fastqc_reads"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 40 × 3</span></span></span>
<span><span class="co">#&gt;    Samples   read_type   n_reads</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> sample_03 unique     2<span style="text-decoration: underline;">685</span>620.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> sample_03 duplicate 10<span style="text-decoration: underline;">908</span>567.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> sample_02 unique     2<span style="text-decoration: underline;">803</span>309.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> sample_02 duplicate 11<span style="text-decoration: underline;">083</span>659.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> sample_05 unique     3<span style="text-decoration: underline;">006</span>011.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> sample_05 duplicate 12<span style="text-decoration: underline;">419</span>121.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> sample_12 unique     3<span style="text-decoration: underline;">186</span>383.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> sample_12 duplicate 12<span style="text-decoration: underline;">738</span>310.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> sample_19 unique     2<span style="text-decoration: underline;">725</span>919.</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> sample_19 duplicate 14<span style="text-decoration: underline;">029</span>645.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 30 more rows</span></span></span></code></pre></div>
</div>
<div class="tab-pane" id="star" role="tabpanel" aria-labelledby="star-tab">

<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_plot_output</span><span class="op">[[</span><span class="st">"data"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"star"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 3</span></span></span>
<span><span class="co">#&gt;    Samples   read_type   n_reads</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> sample_03 unique      6<span style="text-decoration: underline;">352</span>702</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> sample_03 multimapped 3<span style="text-decoration: underline;">876</span>206</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> sample_03 too many       <span style="text-decoration: underline;">2</span>908</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> sample_03 too short   3<span style="text-decoration: underline;">354</span>213</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> sample_03 unmapped       <span style="text-decoration: underline;">8</span>158</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> sample_02 unique      6<span style="text-decoration: underline;">489</span>051</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> sample_02 multimapped 3<span style="text-decoration: underline;">880</span>477</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> sample_02 too many       <span style="text-decoration: underline;">2</span>416</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> sample_02 too short   3<span style="text-decoration: underline;">509</span>469</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> sample_02 unmapped       <span style="text-decoration: underline;">5</span>555</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span></code></pre></div>
</div>
<div class="tab-pane" id="htseq" role="tabpanel" aria-labelledby="htseq-tab">

<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_plot_output</span><span class="op">[[</span><span class="st">"data"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"htseq"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 100 × 3</span></span></span>
<span><span class="co">#&gt;    Samples   read_type  n_reads</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> sample_03 assigned   4<span style="text-decoration: underline;">724</span>190</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> sample_03 ambiguous  1<span style="text-decoration: underline;">228</span>684</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> sample_03 not unique 3<span style="text-decoration: underline;">876</span>206</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> sample_03 no feature  <span style="text-decoration: underline;">399</span>828</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> sample_03 low aQual        0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> sample_02 assigned   4<span style="text-decoration: underline;">831</span>854</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> sample_02 ambiguous  1<span style="text-decoration: underline;">243</span>270</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> sample_02 not unique 3<span style="text-decoration: underline;">880</span>477</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> sample_02 no feature  <span style="text-decoration: underline;">413</span>927</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> sample_02 low aQual        0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 90 more rows</span></span></span></code></pre></div>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="tr_sort_alphanum">tr_sort_alphanum<a class="anchor" aria-label="anchor" href="#tr_sort_alphanum"></a>
</h2>
<p>Sort a column of alphanumeric strings in (non-binary) numerical order
given an input data frame and desired column. You can use the column
name or index, and its compatible with pipes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df_unsorted</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  colA <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"a11"</span>, <span class="st">"a1"</span>, <span class="st">"b1"</span>, <span class="st">"a2"</span><span class="op">)</span>,</span>
<span>  colB <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/tr_sort_alphanum.html">tr_sort_alphanum</a></span><span class="op">(</span>input_df <span class="op">=</span> <span class="va">df_unsorted</span>, sort_col <span class="op">=</span> <span class="st">"colA"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   colA colB</span></span>
<span><span class="co">#&gt; 2   a1    1</span></span>
<span><span class="co">#&gt; 4   a2    2</span></span>
<span><span class="co">#&gt; 1  a11    3</span></span>
<span><span class="co">#&gt; 3   b1    4</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_test_enrichment">tr_test_enrichment<a class="anchor" aria-label="anchor" href="#tr_test_enrichment"></a>
</h2>
<p>Simple wrapper around Fisher’s test for gene enrichment, which
constructs the matrix for you and returns the p value.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">10000</span>, <span class="fl">5000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">de_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">all_genes</span>, <span class="fl">1500</span><span class="op">)</span></span>
<span><span class="va">gene_set</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">all_genes</span>, <span class="fl">100</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/tr_test_enrichment.html">tr_test_enrichment</a></span><span class="op">(</span></span>
<span>  query_set <span class="op">=</span> <span class="va">de_genes</span>, </span>
<span>  enrichment_set <span class="op">=</span> <span class="va">gene_set</span>, </span>
<span>  total_genes <span class="op">=</span> <span class="fl">5000</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.05127792</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_theme">tr_theme<a class="anchor" aria-label="anchor" href="#tr_theme"></a>
</h2>
<p>Clean themes for <a href="https://ggplot2.tidyverse.org/" class="external-link">ggplot2</a>
that improve on the default by increasing font size, changing the
background to white, and adding a border. By default it uses a minimal
grid, but you can easily remove the grid entirely.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="va">basic_box_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">mtcars</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">cyl</span><span class="op">)</span>, <span class="va">mpg</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">basic_box_plot</span> <span class="op">+</span> <span class="fu"><a href="../reference/tr_theme.html">tr_theme</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">basic_box_plot</span> <span class="op">+</span> <span class="fu"><a href="../reference/tr_theme.html">tr_theme</a></span><span class="op">(</span>grid <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="tRavis_files/figure-html/tr_theme-1.png" width="384"><img src="tRavis_files/figure-html/tr_theme-2.png" width="384"></p>
</div>
<div class="section level2">
<h2 id="tr_tidy_gage">tr_tidy_gage<a class="anchor" aria-label="anchor" href="#tr_tidy_gage"></a>
</h2>
<p>Combines the items “greater” and “less” from the list output by <a href="https://bioconductor.org/packages/gage/" class="external-link">gage</a> into a single
tidy data frame (tibble), and provides an option to filter the results
based on q value.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tibble_head</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">x</span>, rownames <span class="op">=</span> <span class="st">"rownames"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">gage_untidy</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"ex_gage_results.rds"</span>, package <span class="op">=</span> <span class="st">"tRavis"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Have a look at the original results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">gage_untidy</span>, <span class="va">tibble_head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $greater</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   rownames                      p.geomean stat.mean  p.val q.val set.size   exp1</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> pae00470 D-Amino acid metabo…    0.063<span style="text-decoration: underline;">0</span>     1.61  0.063<span style="text-decoration: underline;">0</span> 0.924       10 0.063<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> pae01232 Nucleotide metaboli…    0.111      1.24  0.111  0.924       16 0.111 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> pae00640 Propanoate metaboli…    0.167      0.994 0.167  0.924       11 0.167 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> pae00330 Arginine and prolin…    0.179      0.937 0.179  0.924       14 0.179 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> pae03010 Ribosome                0.197      0.862 0.197  0.924       19 0.197 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> pae01210 2-Oxocarboxylic aci…    0.250      0.684 0.250  0.924       12 0.250 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $less</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 7</span></span></span>
<span><span class="co">#&gt;   rownames                      p.geomean stat.mean  p.val q.val set.size   exp1</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> pae00650 Butanoate metabolism    0.076<span style="text-decoration: underline;">4</span>    -<span style="color: #BB0000;">1.48</span>  0.076<span style="text-decoration: underline;">4</span> 0.680       13 0.076<span style="text-decoration: underline;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> pae01230 Biosynthesis of ami…    0.077<span style="text-decoration: underline;">3</span>    -<span style="color: #BB0000;">1.44</span>  0.077<span style="text-decoration: underline;">3</span> 0.680       45 0.077<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> pae00260 Glycine, serine and…    0.087<span style="text-decoration: underline;">7</span>    -<span style="color: #BB0000;">1.39</span>  0.087<span style="text-decoration: underline;">7</span> 0.680       16 0.087<span style="text-decoration: underline;">7</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> pae01100 Metabolic pathways      0.097<span style="text-decoration: underline;">3</span>    -<span style="color: #BB0000;">1.30</span>  0.097<span style="text-decoration: underline;">3</span> 0.680      269 0.097<span style="text-decoration: underline;">3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> pae00630 Glyoxylate and dica…    0.118     -<span style="color: #BB0000;">1.23</span>  0.118  0.680       11 0.118 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> pae01120 Microbial metabolis…    0.160     -<span style="color: #BB0000;">0.999</span> 0.160  0.680       77 0.160 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $stats</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   rownames                                 stat.mean  exp1</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> pae00470 D-Amino acid metabolism             1.61  1.61 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> pae01232 Nucleotide metabolism               1.24  1.24 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> pae00640 Propanoate metabolism               0.994 0.994</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> pae00330 Arginine and proline metabolism     0.937 0.937</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> pae03010 Ribosome                            0.862 0.862</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> pae01210 2-Oxocarboxylic acid metabolism     0.684 0.684</span></span>
<span></span>
<span><span class="fu"><a href="../reference/tr_tidy_gage.html">tr_tidy_gage</a></span><span class="op">(</span><span class="va">gage_untidy</span>, qval <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 70 × 7</span></span></span>
<span><span class="co">#&gt;    pathway                      p_geomean stat_mean  p_val q_val set_size   exp1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> pae00470 D-Amino acid metab…    0.063<span style="text-decoration: underline;">0</span>     1.61  0.063<span style="text-decoration: underline;">0</span> 0.924       10 0.063<span style="text-decoration: underline;">0</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> pae01232 Nucleotide metabol…    0.111      1.24  0.111  0.924       16 0.111 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> pae00640 Propanoate metabol…    0.167      0.994 0.167  0.924       11 0.167 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> pae00330 Arginine and proli…    0.179      0.937 0.179  0.924       14 0.179 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> pae03010 Ribosome               0.197      0.862 0.197  0.924       19 0.197 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> pae01210 2-Oxocarboxylic ac…    0.250      0.684 0.250  0.924       12 0.250 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> pae02040 Flagellar assembly     0.267      0.635 0.267  0.924       11 0.267 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> pae00620 Pyruvate metabolism    0.302      0.526 0.302  0.924       13 0.302 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> pae02024 Quorum sensing         0.327      0.452 0.327  0.924       29 0.327 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> pae00405 Phenazine biosynth…    0.367      0.345 0.367  0.924       10 0.367 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 60 more rows</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="tr_trunc_neatly">tr_trunc_neatly<a class="anchor" aria-label="anchor" href="#tr_trunc_neatly"></a>
</h2>
<p>Simple function to truncate long strings without breaking them in the
middle of a word. Useful for trimming long axis labels in a plot.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tr_trunc_neatly.html">tr_trunc_neatly</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="st">"This is a long string that we want to break neatly"</span>,</span>
<span>  l <span class="op">=</span> <span class="fl">40</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "This is a long string that we want to..."</span></span></code></pre></div>
<p>It’s can also be used inside of a <code>mutate</code> call:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ex_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  col1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>  col2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"This is a pretty long string"</span>,</span>
<span>    <span class="st">"This string is actually a bit longer"</span>,</span>
<span>    <span class="st">"Here is the longest string of them all, just!"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>  <span class="va">ex_df</span>,</span>
<span>  col3 <span class="op">=</span> <span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map_chr</a></span><span class="op">(</span><span class="va">col2</span>, <span class="op">~</span><span class="fu"><a href="../reference/tr_trunc_neatly.html">tr_trunc_neatly</a></span><span class="op">(</span><span class="va">.x</span>, l <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;   col1                                          col2                   col3</span></span>
<span><span class="co">#&gt; 1    1                  This is a pretty long string    This is a pretty...</span></span>
<span><span class="co">#&gt; 2    2          This string is actually a bit longer      This string is...</span></span>
<span><span class="co">#&gt; 3    3 Here is the longest string of them all, just! Here is the longest...</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ggplot2_3.5.1               DESeq2_1.44.0              </span></span>
<span><span class="co">#&gt;  [3] SummarizedExperiment_1.34.0 Biobase_2.64.0             </span></span>
<span><span class="co">#&gt;  [5] MatrixGenerics_1.16.0       matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.56.1        GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.50.0         tRavis_1.6.0               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.1        dplyr_1.1.4             farver_2.1.2           </span></span>
<span><span class="co">#&gt;  [4] fastmap_1.2.0           janitor_2.2.0           digest_0.6.36          </span></span>
<span><span class="co">#&gt;  [7] timechange_0.3.0        lifecycle_1.0.4         magrittr_2.0.3         </span></span>
<span><span class="co">#&gt; [10] compiler_4.4.1          rlang_1.1.4             sass_0.4.9             </span></span>
<span><span class="co">#&gt; [13] tools_4.4.1             utf8_1.2.4              yaml_2.3.9             </span></span>
<span><span class="co">#&gt; [16] knitr_1.48              S4Arrays_1.4.1          labeling_0.4.3         </span></span>
<span><span class="co">#&gt; [19] bit_4.0.5               curl_5.2.1              DelayedArray_0.30.1    </span></span>
<span><span class="co">#&gt; [22] plyr_1.8.9              abind_1.4-5             BiocParallel_1.38.0    </span></span>
<span><span class="co">#&gt; [25] withr_3.0.0             purrr_1.0.2             desc_1.4.3             </span></span>
<span><span class="co">#&gt; [28] grid_4.4.1              fansi_1.0.6             colorspace_2.1-0       </span></span>
<span><span class="co">#&gt; [31] scales_1.3.0            cli_3.6.3               rmarkdown_2.27         </span></span>
<span><span class="co">#&gt; [34] crayon_1.5.3            ragg_1.3.2              generics_0.1.3         </span></span>
<span><span class="co">#&gt; [37] httr_1.4.7              tzdb_0.4.0              cachem_1.1.0           </span></span>
<span><span class="co">#&gt; [40] stringr_1.5.1           zlibbioc_1.50.0         parallel_4.4.1         </span></span>
<span><span class="co">#&gt; [43] XVector_0.44.0          vctrs_0.6.5             Matrix_1.7-0           </span></span>
<span><span class="co">#&gt; [46] jsonlite_1.8.8          hms_1.1.3               bit64_4.0.5            </span></span>
<span><span class="co">#&gt; [49] ggrepel_0.9.5           systemfonts_1.1.0       locfit_1.5-9.10        </span></span>
<span><span class="co">#&gt; [52] tidyr_1.3.1             jquerylib_0.1.4         glue_1.7.0             </span></span>
<span><span class="co">#&gt; [55] pkgdown_2.1.0           codetools_0.2-20        lubridate_1.9.3        </span></span>
<span><span class="co">#&gt; [58] stringi_1.8.4           gtable_0.3.5            UCSC.utils_1.0.0       </span></span>
<span><span class="co">#&gt; [61] munsell_0.5.1           tibble_3.2.1            pillar_1.9.0           </span></span>
<span><span class="co">#&gt; [64] htmltools_0.5.8.1       GenomeInfoDbData_1.2.12 R6_2.5.1               </span></span>
<span><span class="co">#&gt; [67] textshaping_0.4.0       vroom_1.6.5             evaluate_0.24.0        </span></span>
<span><span class="co">#&gt; [70] lattice_0.22-6          highr_0.11              readr_2.1.5            </span></span>
<span><span class="co">#&gt; [73] snakecase_0.11.1        bslib_0.7.0             Rcpp_1.0.13            </span></span>
<span><span class="co">#&gt; [76] SparseArray_1.4.8       xfun_0.46               fs_1.6.4               </span></span>
<span><span class="co">#&gt; [79] forcats_1.0.0           pkgconfig_2.0.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Travis Blimkie.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
