#' Create ggplot2 figures from MultiQC results
#'
#' @param directory Folder containing all the data files generated by MultiQC,
#'   e.g. "multiqc_data/"
#' @param type Type of plot to produce, either "bar" (default) or "box"
#' @param add_points Logical: When making a box plot, should individual samples
#'   be plotted as points? Defaults to TRUE.
#' @param font_size Base font size (defaults to 18)
#' @param threshold_line Provide a number to draw a line at the indicated number
#'   of reads for FastQC read, STAR, and HTSeq plots. Defaults to 10e6; set to
#'   NULL to disable.
#' @param threshold_line_colour Colour for the threshold line ("#EE2C2C")
#' @param threshold_line_type Type of threshold line to draw ("dashed"). See
#'   `?aes_linetype_size_shape` for available options.
#' @param threshold_line_size Size of threshold line (1)
#' @param limits Override the upper limit of FastQC read, STAR, and HTSeq bar
#'   plots. Supply a single number to give all three plots the same limit, or a
#'   vector of three values to modify each individually. Defaults to NULL, which
#'   sets automatic limits.
#'
#' @return A list with elements "plot" containing the `ggplot` objects, and
#'   "data" containing all the underlying data
#' @export
#'
#' @import dplyr
#' @import ggplot2
#' @importFrom forcats fct_inorder
#' @importFrom ggrepel geom_label_repel
#' @importFrom janitor clean_names
#' @importFrom readr cols read_delim
#' @importFrom stringr str_remove str_replace_all str_wrap
#' @importFrom tidyr pivot_longer
#'
#' @description Creates four ggplot2 figures from MultiQC results, and returns
#'   the data along with the plot objects. The plots are for FastQC (read counts
#'   and Phred scores), STAR, and HTSeq. See Details for more information.
#'
#' @details For the Phred scores, one must open the MultiQC HTML report, and
#'   export the data for "fastqc_per_base_sequence_quality_plot" as a tab-
#'   delimited file (TSV), placing it inside the same directory as the rest. If
#'   there are too many samples, the data gets saved in
#'   "mqc_fastqc_per_base_sequence_quality_plot_1.txt"; this file is also
#'   checked for automatically.
#'
#'   Note that the "limits" argument only applies to bar plots - it has no
#'   effect on box plots.
#'
#'   If the data is paired end (i.e. there is R1 and R2 for each sample), each
#'   read will be plotted seperately in the FastQC Phred score and read number
#'   plots. If the samples are named with "R1" but no "R2" is found, the "R1"
#'   will be removed from the sample names.
#'
#' @references <https://multiqc.info/>
#' @seealso <https://www.github.com/travis-m-blimkie/tRavis>
#'
#' @examples
#' if (FALSE) tr_qc_plots("multiqc_data")
#'
tr_qc_plots <- function(
    directory,
    type = "bar",
    add_points = TRUE,
    font_size = 18,
    threshold_line = 10e6,
    threshold_line_colour = "#EE2C2C",
    threshold_line_type = "dashed",
    threshold_line_size = 1,
    limits = NULL
) {

  stopifnot("Argument 'type' must be either 'bar' or 'box'." =
              type %in% c("bar", "box"))


  # Setup -----------------------------------------------------------------

  file_phred_scores <-
    file.path(directory, "fastqc_per_base_sequence_quality_plot.tsv")
  file_phred_scores_alt <-
    file.path(directory, "mqc_fastqc_per_base_sequence_quality_plot_1.txt")

  file_fastqc_reads <- file.path(directory, "multiqc_fastqc.txt")
  file_star <- file.path(directory, "multiqc_star.txt")
  file_htseq <- file.path(directory, "multiqc_htseq.txt")

  draw_line <- ifelse(!is.null(threshold_line), TRUE, FALSE)

  dashed_vline <- geom_vline(
    xintercept = threshold_line,
    linetype = threshold_line_type,
    colour = threshold_line_colour,
    linewidth = threshold_line_size
  )

  dashed_hline <- geom_hline(
    yintercept = threshold_line,
    linetype = threshold_line_type,
    colour = threshold_line_colour,
    linewidth = threshold_line_size
  )

  colour_keys <- list(
    "fastqc" = c(
      "unique" = "#9AC0CD",
      "duplicate" = "#7F7F7F"
    ),
    "star" = c(
      "unique" = "#104E8B",
      "multimapped" = "#7CB5EC",
      "too many" = "#F7A35C",
      "too short" = "#F08080",
      "unmapped" = "#7F0000"
    ),
    "htseq" = c(
      "assigned" = "#7CB5EC",
      "ambiguous" = "#434348",
      "not unique" = "#90ED7D",
      "no feature" = "#F7A35C",
      "low aQual" = "#8085E9"
    )
  )

  output_list <- list()


  # Phred scores ----------------------------------------------------------

  if (any(file.exists(file_phred_scores, file_phred_scores_alt))) {

    if (file.exists(file_phred_scores)) {
      phred_1 <- read_delim(
        file = file_phred_scores,
        delim = "\t",
        col_types = cols()
      )

      phred_2 <- phred_1 %>%
        pivot_longer(
          -`Position (bp)`,
          names_to = "sample",
          values_to = "phred_score"
        ) %>%
        rename("position" = `Position (bp)`)

    } else {
      phred_1 <- read_delim(
        file = file_phred_scores_alt,
        delim = "\t",
        col_types = cols()
      )

      phred_2 <- phred_1 %>%
        pivot_longer(
          !Sample,
          names_to = "position",
          values_to = "phred_score"
        ) %>%
        mutate(position = as.numeric(position)) %>%
        rename("sample" = Sample)
    }

    bad_samples <- phred_2 %>%
      filter(phred_score < 30) %>%
      distinct(sample, .keep_all = TRUE) %>%
      mutate(qc = sample)

    phred_3 <- left_join(
      phred_2,
      bad_samples,
      by = c("position", "sample", "phred_score")
    ) %>%
      relocate(sample, position, phred_score, qc) %>%
      mutate(sample = str_remove(string = sample, pattern = "_?(r|R)[0-9]_?"))

    max_phred <- round_any(
      x = max(phred_3$phred_score),
      accuracy = 5,
      f = ceiling
    )

    line_alpha <- ifelse(length(unique(phred_3$sample)) > 20, 0.3, 1)

    plot_phred_scores <-
      ggplot(phred_3, aes(position, phred_score, group = sample)) +
      geom_line(alpha = line_alpha) +
      geom_hline(
        yintercept = 30,
        linetype = "dashed",
        colour = "#00CD66",
        linewidth = 1
      ) +
      geom_label_repel(
        aes(label = qc),
        size = 4,
        min.segment.length = 0,
        show.legend = FALSE,
        na.rm = TRUE,
        seed = 1
      ) +
      scale_x_continuous(expand = expansion(mult = 0.02)) +
      scale_y_continuous(limits = c(0, max_phred)) +
      labs(
        x = "Position (bp)",
        y = "Phred score",
        title = "FastQC: Mean quality scores"
      ) +
      tr_theme(base_size = font_size)

    output_list$plots$phred_scores <- plot_phred_scores
    output_list$data$phred_scores <- select(phred_3, !qc)
  } else {
    message(
      "Required file 'fastqc_per_base_sequence_quality_plot.tsv' or ",
      "'mqc_fastqc_per_base_sequence_quality_plot_1.txt' not found. To get ",
      "this plot, open the MultiQC HTML report and export the data for the ",
      "'FastQC per-base sequence quality' as a tab-delimited file (TSV), ",
      "saving into the same data directory."
    )
  }


  # FastQC reads ----------------------------------------------------------

  if (file.exists(file_fastqc_reads)) {

    fastqc_1 <-
      read_delim(file_fastqc_reads, delim = "\t", col_types = cols()) %>%
      clean_names()

    fastqc_2 <- fastqc_1 %>%
      mutate(
        unique = total_sequences * (total_deduplicated_percentage / 100),
        duplicate = total_sequences - unique
      ) %>%
      select(sample, unique, duplicate, total_sequences) %>%
      arrange(total_sequences) %>%
      mutate(sample = fct_inorder(sample))

    fastqc_3 <- fastqc_2 %>%
      select(-total_sequences) %>%
      pivot_longer(
        unique:duplicate,
        names_to = "read_type",
        values_to = "n_reads"
      )

    found_r1 <- any(grepl(x = tolower(fastqc_3$sample), pattern = "r1"))
    found_r2 <- any(grepl(x = tolower(fastqc_3$sample), pattern = "r2"))

    if (found_r1 & !found_r2) { # R1 but no R2
      fastqc_4 <- fastqc_3 %>%
        mutate(sample = fct_inorder(str_remove(sample, "_?(r|R)1_?")))
    } else {
      fastqc_4 <- fastqc_3 %>%
        mutate(sample = fct_inorder(sample))
    }

    rounded_max_fastqc <-
      if (is.null(limits)) {
        get_rounded_max(fastqc_4)
      } else if (length(limits) == 1) {
        limits
      } else if (length(limits) == 3) {
        limits[1]
      } else {
        stop(
          "Argument 'limits' must be NULL, a single value, or a numeric ",
          "vector with length 3."
        )
      }

    plot_fastqc_reads <- if (type == "bar") {
      ggplot(fastqc_4, aes(n_reads, sample, fill = read_type)) +
        geom_col() +
        {if (draw_line) dashed_vline} +
        scale_fill_manual(values = colour_keys$fastqc) +
        scale_x_continuous(
          expand = expansion(mult = c(0, 0.1)),
          limits = c(0, rounded_max_fastqc),
          labels = ~.x / 1e6
        ) +
        labs(
          x = "Reads (M)",
          y = NULL,
          title = "FastQC: Sequence counts",
          fill = "Read type"
        ) +
        tr_theme(base_size = font_size) +
        theme(
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank()
        )

    } else if (type == "box") {

      if (add_points) {
        plot_fastqc_base <-
          ggplot(fastqc_4, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(outlier.shape = NA) +
          geom_point(
            pch = 21,
            size = 2,
            alpha = 0.8,
            position = position_jitter(width = 0.25, height = 0, seed = 1)
          )
      } else {
        plot_fastqc_base <-
          ggplot(fastqc_4, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(
            outlier.shape = 21,
            outlier.size = 2,
            outlier.alpha = 0.8,
            outlier.fill = NULL
          )
      }

      plot_fastqc_base +
        scale_y_continuous(labels = ~.x/1e6) +
        scale_fill_manual(values = colour_keys$fastqc, guide = NULL) +
        {if (draw_line) dashed_hline} +
        labs(
          x = "Read type",
          y = "Reads (M)",
          title = "FastQC: Sequence counts"
        ) +
        tr_theme(base_size = font_size)
    }

    output_list$plots$fastqc_reads <- plot_fastqc_reads
    output_list$data$fastqc_reads <- fastqc_4
  } else {
    message(
      "No data found for FastQC reads; check that 'multiqc_fastqc.txt' exists."
    )
  }


  # STAR ------------------------------------------------------------------

  if (file.exists(file_star)) {

    star_1 <- read_delim(
      file = file_star,
      delim = "\t",
      col_types = cols()
    ) %>%
      clean_names()

    star_2 <- star_1 %>%
      arrange(total_reads) %>%
      mutate(sample = fct_inorder(sample)) %>%
      select(
        sample,
        "unique" = uniquely_mapped,
        multimapped,
        "too_many" = multimapped_toomany,
        "too_short" = unmapped_tooshort,
        "unmapped" = unmapped_other
      )

    star_3 <- star_2 %>%
      pivot_longer(
        -sample,
        names_to = "read_type",
        values_to = "n_reads"
      ) %>%
      mutate(
        read_type = str_replace_all(read_type, pattern = c("_" = " ")),
        read_type = factor(read_type, c(
          "unmapped",
          "too short",
          "too many",
          "multimapped",
          "unique"
        ))
      )

    rounded_max_star <-
      if (is.null(limits)) {
        get_rounded_max(star_3)
      } else if (length(limits) == 1) {
        limits
      } else if (length(limits) == 3) {
        limits[2]
      } else {
        stop(
          "Argument 'limits' must be NULL, a single value, or a numeric ",
          "vector with length 3."
        )
      }

    plot_star <- if (type == "bar") {
      ggplot(star_3, aes(n_reads, sample, fill = read_type)) +
        geom_col() +
        {if (draw_line) dashed_vline} +
        scale_x_continuous(
          expand = expansion(mult = c(0, 0.1)),
          limits = c(0, rounded_max_star),
          labels = ~.x / 1e6
        ) +
        scale_fill_manual(values = colour_keys$star) +
        labs(
          x = "Reads (M)",
          y = NULL,
          title = "STAR: Alignment scores",
          fill = "Read type"
        ) +
        tr_theme(base_size = font_size) +
        theme(
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank()
        )

    } else if (type == "box") {

      if (add_points) {
        plot_star_base <-
          ggplot(star_3, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(outlier.shape = NA) +
          geom_point(
            pch = 21,
            size = 2,
            alpha = 0.8,
            position = position_jitter(width = 0.25, height = 0, seed = 1)
          )
      } else {
        plot_star_base <-
          ggplot(star_3, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(
            outlier.shape = 21,
            outlier.size = 2,
            outlier.alpha = 0.8,
            outlier.fill = NULL
          )
      }

      plot_star_base +
        scale_x_discrete(labels = ~str_wrap(.x, width = 3)) +
        scale_y_continuous(labels = ~.x/1e6) +
        scale_fill_manual(values = colour_keys$star, guide = NULL) +
        {if (draw_line) dashed_hline} +
        labs(
          x = "Read type",
          y = "Reads (M)",
          title = "STAR: Alignment scores"
        ) +
        tr_theme(base_size = font_size)
    }

    output_list$plots$star <- plot_star
    output_list$data$star <- star_3
  } else {
    message("No data found for STAR; check that 'multiqc_star.txt' exists.")
  }


  # HTSeq -----------------------------------------------------------------

  if (file.exists(file_htseq)) {

    htseq_1 <- read_delim(
      file = file_htseq,
      delim = "\t",
      col_types = cols()
    ) %>%
      clean_names()

    htseq_2 <- htseq_1 %>%
      arrange(total_count) %>%
      mutate(
        sample = str_remove(sample, ".count$"),
        sample = fct_inorder(sample)
      ) %>%
      select(
        sample,
        assigned,
        ambiguous,
        "not_unique" = alignment_not_unique,
        no_feature,
        "low_aQual" = too_low_a_qual
      )

    htseq_3 <- htseq_2 %>%
      pivot_longer(
        -sample,
        names_to = "read_type",
        values_to = "n_reads"
      ) %>%
      mutate(
        read_type = str_replace_all(read_type, pattern = c("_" = " ")),
        read_type = factor(read_type, c(
          "low aQual",
          "no feature",
          "not unique",
          "ambiguous",
          "assigned"
        ))
      )

    rounded_max_htseq <-
      if (is.null(limits)) {
        get_rounded_max(htseq_3)
      } else if (length(limits) == 1) {
        limits
      } else if (length(limits) == 3) {
        limits[3]
      } else {
        stop(
          "Argument 'limits' must be NULL, a single value, or a numeric ",
          "vector with length 3."
        )
      }

    plot_htseq <- if (type == "bar") {
      ggplot(htseq_3, aes(n_reads, sample, fill = read_type)) +
        geom_col() +
        {if (draw_line) dashed_vline} +
        scale_x_continuous(
          expand = expansion(mult = c(0, 0.1)),
          limits = c(0, rounded_max_htseq),
          labels = ~.x / 1e6
        ) +
        scale_fill_manual(values = colour_keys$htseq) +
        labs(
          x = "Reads (M)",
          y = NULL,
          title = "HTSeq: Count assignments",
          fill = "Read type"
        ) +
        tr_theme(base_size = font_size) +
        theme(
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank()
        )

    } else if (type == "box") {

      if (add_points) {
        plot_htseq_base <-
          ggplot(htseq_3, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(outlier.shape = NA) +
          geom_point(
            pch = 21,
            size = 2,
            alpha = 0.8,
            position = position_jitter(width = 0.25, height = 0, seed = 1)
          )
      } else {
        plot_htseq_base <-
          ggplot(htseq_3, aes(read_type, n_reads, fill = read_type)) +
          geom_boxplot(
            outlier.shape = 21,
            outlier.size = 2,
            outlier.alpha = 0.8,
            outlier.fill = NULL
          )
      }

      plot_htseq_base +
        scale_x_discrete(labels = ~str_wrap(.x, width = 3)) +
        scale_y_continuous(labels = ~.x/1e6) +
        scale_fill_manual(values = colour_keys$htseq, guide = NULL) +
        {if (draw_line) dashed_hline} +
        labs(
          x = "Read type",
          y = "Reads (M)",
          title = "HTSeq: Count assignments"
        ) +
        tr_theme(base_size = font_size)
    }

    output_list$plots$htseq <- plot_htseq
    output_list$data$htseq <- htseq_3
  } else {
    message("No data found for HTSeq; check that 'multiqc_htseq.txt' exists.")
  }

  return(output_list)
}
